core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -81,6 +85,7 @@
 
   private final ClientTransportFactory transportFactory;
   private final ExecutorService executor;
+  private volatile ScheduledExecutorService deadlineCancellationExecutor;
```

Comment:
I see now it can't be `final`, but it doesn't need to be volatile as it would still only be accessed during the constructor or with the intrinsic lock held.